File: autojs/src/main/java/com/baidu/paddle/lite/demo/ocr/OcrResult.java
package com.baidu.paddle.lite.demo.ocr;

import android.graphics.Rect;
import android.os.Parcelable;

public class OcrResult implements Comparable<OcrResult> {
    public float confidence;
    public float preprocessTime;
    public float inferenceTime;
    public String words;
    public Rect bounds;
    public RectLocation location;


    @Override
    public int compareTo(OcrResult s) {
        int deviation = Math.max(this.location.height / 2, s.location.height / 2);
        if (Math.abs((this.bounds.top + this.bounds.bottom) / 2 - (s.bounds.top + s.bounds.bottom) / 2) < deviation) {
            return this.bounds.left - s.bounds.left;
        } else {
            return this.bounds.bottom - s.bounds.bottom;
        }
    }


    public static class RectLocation {
        public int left;
        public int top;
        public int width;
        public int height;

        public RectLocation(int left, int top, int width, int height) {
            this.left = left;
            this.top = top;
            this.width = width;
            this.height = height;
        }
    }
}

File: paddleocr/src/main/java/com/baidu/paddle/lite/demo/ocr/OcrResult.kt
package com.baidu.paddle.lite.demo.ocr

import android.graphics.Rect
import kotlin.math.abs

data class OcrResult(
    @JvmField
    val confidence: Float,
    @JvmField
    val preprocessTime: Float,
    @JvmField
    val inferenceTime: Float,
    @JvmField
    val text: String,
    @JvmField
    val bounds: Rect,
) : Comparable<OcrResult> {

    @Deprecated("use text", ReplaceWith("text"))
    @JvmField
    val words: String = text

    @Deprecated("use text", ReplaceWith("text"))
    fun getWords() = text

    @Deprecated("use confidence", ReplaceWith("confidence"))
    fun getConfidence() = confidence

    @Deprecated("use preprocessTime", ReplaceWith("preprocessTime"))
    fun getPreprocessTime() = preprocessTime

    @Deprecated("use inferenceTime", ReplaceWith("inferenceTime"))
    fun getInferenceTime() = inferenceTime

    @Deprecated("use bounds", ReplaceWith("bounds"))
    fun getBounds() = bounds

    override fun compareTo(other: OcrResult): Int {
        val deviation = (bounds.height() / 2f).coerceAtLeast(other.bounds.height() / 2f)
        return if (abs((bounds.top + bounds.bottom) / 2f - (other.bounds.top + other.bounds.bottom) / 2f) < deviation) {
            bounds.left - other.bounds.left
        } else {
            bounds.bottom - other.bounds.bottom
        }
    }

}